### ISSUE TITLE
Drag objects to left/ right edge to move object across workspace

### ISSUE THREAD

**Issue by kdwk** [elementary/gala#1299]
### Problem

Currently, a user has no way to drag and drop items, be it files or text, across workspaces. They must bring whichever destination window they want to the current workspace and then perform the drag action. 

### Proposal

Dragging items to the left/ right edge of the screen and hovering should automatically switch workspaces and carry the item across the left/ right workspace. This makes for a fluid experience for multitasking across workspaces. 

For cohesiveness, dragging windows across to workspaces should also work the same way, and snapping windows to left/ right of the screen can be done by dragging the window to the left/ right third of wingpanel, and maximizing the window can be done by dragging the window to the middle third of wingpanel.

**Comment by cassidyjames** [comment]
>snapping windows to left/ right of the screen can be done by dragging the window to the left/ right third of wingpanel, and maximizing the window can be done by dragging the window to the middle third of wingpanel

Note that this would prevent window tiling from working on external displays.

**Comment by RustoMCSpit** [comment]
i disagree, dragging windows to the side or corner should just tile them.

**Comment by lenemter** [comment]
Fixed in https://github.com/elementary/gala/pull/2275
